### ISSUE TITLE
Device request: Eeese Otto dehumidifier

### ISSUE THREAD

**Issue by cnrd** [make-all/tuya-local#306]
Hi

EDIT: See next comment for a partially working config.

Tuya product: Not sure where to get this?
OEM-name: Eesee Adam
URL: https://eeese-aircare.com/eeese_sortiment/dehumidifier/otto-dehumidifier-60-m2-suitibile-for-wet-rooms-wifi-google-assistant-35-db_1

Tried adding my Eeese Otto dehumidifier, I had expected it to work with the same config as the adam (https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/eesee_adam_dehumidifier.yaml)
It was added in this PR: https://github.com/make-all/tuya-local/issues/180

There is a spelling mistake in that one btw, it should be [Eeese not Eesee](http://eeese-aircare.com/) 😄

Information from Hass
```
Device matches mirabella_genio_usb with quality of 22%. DPS: {'1': True, '2': 25, '5': 'Manual', '6': 55, '8': True, '10': False, '16': False, '17': 'cancel', '19': 0, 'updated_at': 1670694781.640623}
```
Information from Tuya device specification.
```
{
  "result": {
    "category": "cs",
    "functions": [
      {
        "code": "switch",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "dehumidify_set_value",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":25,\"max\":80,\"scale\":0,\"step\":5}"
      },
      {
        "code": "swing",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "anion",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "child_lock",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "countdown_set",
        "type": "Enum",
        "values": "{\"range\":[\"cancel\",\"1h\",\"2h\",\"3h\"]}"
      }
    ],
    "status": [
      {
        "code": "switch",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "dehumidify_set_value",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":25,\"max\":80,\"scale\":0,\"step\":5}"
      },
      {
        "code": "humidity_indoor",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      },
      {
        "code": "swing",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "anion",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "child_lock",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "countdown_set",
        "type": "Enum",
        "values": "{\"range\":[\"cancel\",\"1h\",\"2h\",\"3h\"]}"
      },
      {
        "code": "fault",
        "type": "Bitmap",
        "values": "{\"label\":[\"E1\",\"E2\"]}"
      }
    ]
  },
  "success": true,
  "t": 1670695212706,
  "tid": "7e41b5eb78b411edb7f63a1ad91050e8"
}
```

It seems like the dp_ids have been removed, so I have gotten them using the zigbee2mqtt "log" way.
```
Child lock: 16
Power: 1
Timer off: 17
Dehumidify: 2
Error code: 19
Working mode: 5
Indoor Humidity: 6
Swing: 8
Anion: 10
```
These were all the types of device logs listed.

Using diagnostics in hass I was able to get the following values for dp_id 5:
```
Manual
Clothdrying
SLeeping (Yes with that capitalization)
```

Please let me know if there is anything else I can supply.

**Comment by cnrd** [comment]
I decided to cut up the existing Eeese Adam config and created the following (partially working) config

```
name: Eesee Otto dehumidifier
primary_entity:
  entity: humidifier
  class: dehumidifier
  dps:
    - id: 1
      name: switch
      type: boolean
      mapping:
        - dps_val: false
          icon: "mdi:air-humidifier-off"
          icon_priority: 1
        - dps_val: true
          icon: "mdi:air-humidifier"
          icon_priority: 4
    - id: 2
      name: humidity
      type: integer
      range:
        min: 25
        max: 80
      mapping:
        - step: 5
    - id: 8
      name: swing
      type: boolean
    - id: 10
      name: anion
      type: boolean
    - id: 19
      type: bitfield
      name: error
      mapping:
        - dps_val: 1
          icon: "mdi:cup-water"
          icon_priority: 2
      hidden: true
secondary_entities:
  - entity: fan
    dps:
      - id: 1
        type: boolean
        name: switch
  - entity: sensor
    name: Current humidity
    class: humidity
    dps:
      - id: 6
        type: integer
        name: sensor
        unit: "%"
        class: measurement
  - entity: binary_sensor
    name: Tank
    class: problem
    category: diagnostic
    dps:
      - id: 19
        type: bitfield
        name: sensor
        mapping:
          - dps_val: 0
            value: false
          - value: true
  - entity: lock
    name: Child lock
    category: config
    dps:
      - id: 16
        type: boolean
        name: lock
        mapping:
          - dps_val: true
            icon: "mdi:hand-back-right-off"
          - dps_val: false
            icon: "mdi:hand-back-right"
  - entity: select
    name: Timer
    icon: "mdi:timer"
    category: config
    dps:
      - id: 17
        type: string
        name: option
        mapping:
          - dps_val: cancel
            value: "Off"
          - dps_val: 1h
            value: 1 hour
          - dps_val: 2h
            value: 2 hours
          - dps_val: 3h
            value: 3 hours
          - dps_val: 4h
            value: 4 hours
          - dps_val: 5h
            value: 5 hours
          - dps_val: 6h
            value: 6 hours
          - dps_val: 7h
            value: 7 hours
          - dps_val: 8h
            value: 8 hours
          - dps_val: 9h
            value: 9 hours
          - dps_val: 10h
            value: 10 hours
          - dps_val: 11h
            value: 11 hours
          - dps_val: 12h
            value: 12 hours
          - dps_val: 13h
            value: 13 hours
          - dps_val: 14h
            value: 14 hours
          - dps_val: 15h
            value: 15 hours
          - dps_val: 16h
            value: 16 hours
          - dps_val: 17h
            value: 17 hours
          - dps_val: 18h
            value: 18 hours
          - dps_val: 19h
            value: 19 hours
          - dps_val: 20h
            value: 20 hours
          - dps_val: 21h
            value: 21 hours
          - dps_val: 22h
            value: 22 hours
          - dps_val: 23h
            value: 23 hours
          - dps_val: 24h
            value: 24 hours
```

Neither swing nor anion shows up as switches, I probably missed something there...

The humidifier has a mode button, which allows me to toggle between normal, clothes dryer and night modes, always in that order. Could that be related to the "Working mode" dp_id?

**Comment by make-all** [comment]
The "name" of the switch dp should be "switch".  Also, the icon-priorities are not needed for the switches since there is only one place where the icons are set, and the icon can be moved up a level since it does not change with the state.  

dp 5 can be implemented in the humidifier entity as

```
    - id: 5
      name: mode
      type: string
      mapping:
        - dps_val: Manual
          value: Manual
        - dps_val: Clothdrying
          value: Dry Clothes
          icon: "mdi:t-shirt-crew"
          icon_priority: 3
        - dps_val: SLeeping
          value: Sleep
          icon: "mdi:sleep"
          icon_priority: 3
```

**Comment by cnrd** [comment]
Just tested all functionality using the file from your PR, everything seems to be working, thanks!

The device does have some quirks (Just wanted to list them in case someone hits them):
- Sending multiple commands in quick succession seems to lock it up for a little while (For example switching Ionizer and swing quickly)
- Sending commands while the child lock is active is not possible, I guess it just emulates button presses, so one has to unlock the child lock as the first command in an automation and relock as the last.

**Comment by make-all** [comment]
I might add your first point to the README, as this is a common limitation of Tuya devices (they seem to be single threaded and get overloaded easily if you try to send more commands when they are still processing the previous one, often crashing the device).  This is mainly a concern for automations where you need to add delays between commands.

The second one seems to be a quirk of this device - most seem to implement the child lock on the device buttons, and still accept remote commands.

**Comment by mrlidstrom** [comment]
I havs added the integration via HACS (Home Assistant 2023.5.3) and found the static local ip, the device-ID and the Local Key. Still my integration does not find the Eeese Otto when I try to add it. What can be done to solve this? I have even tried to delete all .yaml-files except the Otto-file (in folder devices)🤬🤬🤬

**Comment by mrlidstrom** [comment]
Update: this code worked for me (had to erase error parts):

`name: Eeese Otto
primary_entity:
#  entity: switch
  entity: humidifier
  class: dehumidifier
#  class: outlet
  dps:
    - id: 1
      type: boolean
      name: switch
      mapping:
        - dps_val: false
          icon: mdi:air-humidifier-off
          icon_priority: 2
        - dps_val: true
          icon: mdi:air-humidifier
          icon_priority: 4
    - id: 2
      name: humidity
      type: integer
      range:
        min: 25
        max: 80
      mapping:
        - step: 5
    - id: 5
      name: mode
      type: string
      mapping:
        - dps_val: Manual
          value: normal
        - dps_val: Clothdrying
          value: laundry
          icon: "mdi:t-shirt-crew"
          icon_priority: 3
        - dps_val: SLeeping
          value: sleep
          icon: "mdi:sleep"
          icon_priority: 3
secondary_entities:
  - entity: sensor
    name: Current humidity
    class: humidity
    dps:
      - id: 6
        type: integer
        name: sensor
        unit: "%"
        class: measurement
  - entity: switch
    name: Swing
    icon: "mdi:cached"
    category: config
    dps:
      - id: 8
        type: boolean
        name: switch
  - entity: switch
    name: Ionizer
    icon: "mdi:atom-variant"
    category: config
    dps:
      - id: 10
        type: boolean
        name: switch
  - entity: select
    name: Timer
    icon: "mdi:timer"
    category: config
    dps:
      - id: 17
        type: string
        name: option
        mapping:
          - dps_val: cancel
            value: "Off"
          - dps_val: 1h
            value: 1 hour
          - dps_val: 2h
            value: 2 hours
          - dps_val: 3h
            value: 3 hours
          - dps_val: 4h
            value: 4 hours
          - dps_val: 5h
            value: 5 hours
          - dps_val: 6h
            value: 6 hours
          - dps_val: 7h
            value: 7 hours
          - dps_val: 8h
            value: 8 hours
          - dps_val: 9h
            value: 9 hours
          - dps_val: 10h
            value: 10 hours
          - dps_val: 11h
            value: 11 hours
          - dps_val: 12h
            value: 12 hours
          - dps_val: 13h
            value: 13 hours
          - dps_val: 14h
            value: 14 hours
          - dps_val: 15h
            value: 15 hours
          - dps_val: 16h
            value: 16 hours
          - dps_val: 17h
            value: 17 hours
          - dps_val: 18h
            value: 18 hours
          - dps_val: 19h
            value: 19 hours
          - dps_val: 20h
            value: 20 hours
          - dps_val: 21h
            value: 21 hours
          - dps_val: 22h
            value: 22 hours
          - dps_val: 23h
            value: 23 hours
          - dps_val: 24h
            value: 24 hours`

**Comment by Pastar243** [comment]
Hi, 
I´m kind of new to this but I will give it a try and see how it works. I have a eeese luna dehumidifier & air purifier and Tuya Local can´t find it when I try to add it (the alternatives I can choose from does not match). 
I have been trying to follow instructions as far as I understand (getting the dp_id etc), and also It seems to return somewhat the same things that in this setup above for eease otto. 

I get: 
1":"Power"
2":"Humidify Setting"
4":"Mode"
6":"Indoor Humidity",
8":"Swing",
10":"Anion",
17":"Timer Off"
18":"Countdown Left"
19":"Error Code"

Unfortunately I´m not so good at this, but is there anyway I can get support for this device within the capabilities I have (kind of limited) 

Best regards //Patrik

**Comment by Privatecoder** [comment]
Hi Patrik,

this is a bit off-topic but I would try and install the `LocalTuya` integration, enter your tuya-cloud-details and then use its assistant to get the parameters for your device i.e.

1)
<img width="372" alt="image" src="https://github.com/make-all/tuya-local/assets/45964815/1ba283cf-a232-418a-a116-62a3740e683a">
2)
<img width="575" alt="image" src="https://github.com/make-all/tuya-local/assets/45964815/d80a29fe-e774-4543-acd4-5154d278ad71">
3)
<img width="560" alt="image" src="https://github.com/make-all/tuya-local/assets/45964815/585260cf-1ba8-4a9d-81b8-f0d71de7a014">

Then enter these details in this – `Tuya local` – integration and see if it gives you the option to select a device.

Only after that (and selecting one of the existing eeese humidifiers) I would check whats working and what is not.

Cheers

**Comment by Pastar243** [comment]
Thanks for the quick reply. 
I gave it a try and all the data that was shown in local-tuya (host,device id, local key and protocoll version) I already had put in the Tuya local app but it still only gives me opportunity to choose from.

feit_dimmer
co2_box
inkbird_pth9cw_airquality
mirabella_genio_usb
nedis_openclose_sensor
simple_lightbulb
simple_switch

I understand that I might do something wrong here. 
What I could see in Local tuya was. 
![image](https://github.com/make-all/tuya-local/assets/74463632/d89d9705-56be-4051-a295-01c6b11f130a)

**Comment by make-all** [comment]
It is not that you are doing something wrong when configuring, your device is different than the one this issue is for, and is not a supported device.

Rather than trying to get help by commenting on a closed issue for a different device, I suggest that you file a new issue for a "New Device", and fill out the requested log and dps information there.  The log message from Home assistant log is required, and If possible include the "Query Things Data Model" result from iot.tuya.com, as that has required information on ranges and options supported.
